src/pf-touchspin/pf-touchspin.component.js

```diff
@@ -0,0 +1,387 @@
+        return (Math.round(value / this._step) * this._step).toFixed(this._decimals);
+      case 'floor':
+        return (Math.floor(value / this._step) * this._step).toFixed(this._decimals);
+      case 'ceil':
+        return (Math.ceil(value / this._step) * this._step).toFixed(this._decimals);
+      default:
+        return value;
+    }
+  }
+
+  /**
+   * check the value before change in value
+   */
+  _checkValue() {
+    let val, parsedval, returnval;
+
+    val = this.querySelector('input').value;
+
+    if (this._decimals > 0 && val === '.') {
+      return;
+    }
+
+    parsedval = parseFloat(val);
+
+    if (isNaN(parsedval)) {
+      parsedval = 0;
+    }
+
+    returnval = parsedval;
+
+    if (parsedval < this.min) {
+      returnval = this.min;
+    }
+
+    if (parsedval > this.max) {
+      returnval = this.max;
+    }
+
+    returnval = this._stepDivisibility(returnval);
+
+    if (Number(val).toString() !== returnval.toString()) {
+      this.querySelector('input').value = returnval;
+    }
+  }
+
+
+  /**
+   * boost the value
+   *
+   * @param {number} value
+   */
+  _boostedStep(value) {
+    if (!this._booster) {
+      return this._step;
+    }
+    let boosted = Math.pow(2, Math.floor(this.spincount / this._boostat)) * this._step;
+
+    if (this._maxBoostedStep) {
+      if (boosted > this._maxBoostedstep) {
+        boosted = this._maxBoostedStep;
+        value = Math.round((value / boosted)) * boosted;
+      }
+    }
+    return Math.max(this._step, boosted);
+
+  }
+
+  /**
+   * increment input value
+   */
+  _up() {
+    let val, boostedStep;
+
+    this._checkValue();
+
+    val = parseFloat(this.querySelector('input').value);
+    if (isNaN(val)) {
+      val = 0;
+    }
+
+    boostedStep = this._boostedStep(val);
+
+    val = val + boostedStep;
+
+    if (val > this._max) {
+      val = this._max;
+      this.dispatchEvent(new CustomEvent('pf-touchspin.max', {}));
+      this._stop();
+    }
+
+    val = parseFloat(val).toFixed(this._decimals);
+
+    this.querySelector('input').value = val;
+  }
+
+  /**
+   *  decrement input value
+   */
+  _down() {
+    let val, boostedStep;
+
+    this._checkValue();
+
+    val = parseFloat(this.querySelector('input').value);
+    if (isNaN(val)) {
+      val = 0;
+    }
+
+    boostedStep = this._boostedStep(val);
+
+    val = val - boostedStep;
+
+    if (val < this._min) {
+      val = this._min;
+      this.dispatchEvent(new CustomEvent('pf-touchspin.min', {}));
+      this._stop();
+    }
+
+    val = parseFloat(val).toFixed(this._decimals);
+
+    this.querySelector('input').value = val;
+  }
+
+  /**
+   * Decremental spinner
+   *
+   */
+  _downSpin() {
+    let self = this;
+    this._stop();
+
+    this.spincount = 0;
+    this._spinning = 'down';
+
+    this.dispatchEvent(new CustomEvent('pf-touchspin.startspin', {}));
+    this.dispatchEvent(new CustomEvent('pf-touchspin.startdownspin', {}));
+
+    this._downDelayTimeout = setTimeout(function () {
```

Comment:
If we're up for it would an arrow function help eliminate some of the old skool ```self``` references? @priley86 may have already covered this